### PR TITLE
New doc: GDPR support in ASP.NET Core

### LINKED ISSUE #5400: GDPR compliance in ASP.NET Core 2.1+
Document GDPR support

new doc: How to add/delete custom user data to Identity - GDPR #6226

[Barry on GDPR](https://blogs.msdn.microsoft.com/webdev/2018/03/04/asp-net-core-2-1-0-preview1-gdpr-enhancements/)
See aspnet/Security#1561, aspnet/Identity#1341 

[ASP.NET Core 2.1 roadmap blog](https://blogs.msdn.microsoft.com/webdev/2018/02/02/asp-net-core-2-1-roadmap/)

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #5400 

[Review URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/gdpr?view=aspnetcore-2.1&branch=pr-en-us-6516)

### CODE REVIEW

**Comment by Tratcher** [comment]
Recommended Ordering:
-	CookiePolicyOptions and UseCookiePolicy
-	_CookieConsentPartial.cshtml partial view
-	Privacy.cshtml
-	Essential Cookies
-	Tempdata & Session cookies
-	Identity
o	Download
o	Delete
o	Encrypt
o	Etc..

File: aspnetcore/security/gdpr.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+## Tempdata provider cookie is not essential
+
+The [Tempdata provider](xref:fundamentals/app-state#tempdata) cookie is not essential. If tracking is disabled, the Tempdata provider is not functional. To enable the Tempdata provider when tracking is disabled, mark the TempData cookie as essential in `ConfigureServices`:
+
+[!code-csharp[Main](gdpr/sample/RP/Startup.cs?name=snippet1)]
+
+## Essential cookies
+
+If tracking is disabled, only cookies marked essential are sent to the browser. The following code makes a cookie essential:
```
Comment (Tratcher) [inline aspnetcore/security/gdpr.md]: If consent has not been given...

**Comment by blowdart** [comment]
Don't mention encryption, until I have a sample I'm happy with. We'll use that to create a document all on it's own. You could however add a paragraph like the following,

_Some databases or storage mechanisms allow for encryption at rest,  encrypting your stored data with no work needed for any software that accesses the data. This is, by far, the easiest and safest option, let the database manage keys and encryption for you. For example, Microsoft SQL and Azure SQL provide  [Transparent Data Encryption](https://docs.microsoft.com/en-us/sql/relational-databases/security/encryption/transparent-data-encryption?view=sql-server-2017) (TDE), and Azure has encrypted SQL database by default since [May 2017](https://azure.microsoft.com/en-us/updates/newly-created-azure-sql-databases-encrypted-by-default/), as well as encrypting blobs, files, tables and queue storage since [August 2017](https://azure.microsoft.com/en-us/updates/newly-created-azure-sql-databases-encrypted-by-default/). For databases that don't provide built-in encryption at rest you may be able to use disk encryption, such as [Bitlocker](https://docs.microsoft.com/en-us/windows/security/information-protection/bitlocker/bitlocker-how-to-deploy-on-windows-server) to provide the same protections. Linux has encrypted file systems so as [eCryptfs](https://launchpad.net/ecryptfs) and [EncFS](https://github.com/vgough/encfs)._

**Comment by blowdart** [comment]
Should we highlight that the delete and download only touch the default identity data, and needs to be extended out as soon as you start storing other things? Also mention cascading deletes - https://github.com/aspnet/Identity/issues/1797#issuecomment-391052520

**Comment by Rick-Anderson** [comment]
@Tratcher @HaoK @blowdart  GDPR ready for final review.

**Pull request merged**